sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -493,6 +493,13 @@ object DateTimeUtils {
     Decimal(getMicroseconds(micros, zoneId), 8, 6)
   }
 
+  /**
+   * Returns the number of seconds since 1970-01-01 00:00:00-00 (can be negative).
+   */
+  def getSecondsAfterEpoch(micros: Long, zoneId: ZoneId): Double = {
+    micros.toDouble / MICROS_PER_SECOND
```

Comment:
@gengliangwang If you would like to be compatible to PostgreSQL, you need to take the removed implementation:
```scala
/**
   * Returns the number of seconds with fractional part in microsecond precision
   * since 1970-01-01 00:00:00 local time.
   */
  def getEpoch(timestamp: SQLTimestamp, zoneId: ZoneId): Decimal = {
    val offset = SECONDS.toMicros(
      zoneId.getRules.getOffset(microsToInstant(timestamp)).getTotalSeconds)
    val sinceEpoch = timestamp + offset
    Decimal(sinceEpoch, 20, 6)
  }
```
PostgreSQL takes seconds since the **local** epoch 1970-01-01 00:00:00-00 but your implementation calculates seconds since 1970-01-01 00:00:00-00**Z** (in UTC time zone).

Reply:
+1, thanks for pointing it out 👍